platform/security/certificate/security-certificate-generator/src/main/java/org/codice/ddf/security/certificate/generator/PkiTools.java

```diff
@@ -260,6 +263,55 @@ public static X500Name convertDistinguishedName(String... tuples) {
     }
 
     /**
+     * Create an X509 general name based on the specified string which supports a format similar to
+     * OpenSSL X509 configuration as: {@code tag:name} where tag can be one of:
+     * <ul>
+     * <li>email</li>
+     * <li>URI - uniformed resource identifier</li>
+     * <li>RID - registered id</li>
+     * <li>DNS</li>
+     * <li>IP - ip address (V4 or V6)</li>
+     * <li>dirName - directory name</li>
+     * </ul>
+     *
+     * @param name a string representing an X509 general name in an openssl format
+     * @return the corresponding parsed X509 general name
+     * @throws IllegalArgumentException if the string is <code>null</code>, empty or cannot be parsed properly
+     * @see <a href="https://www.openssl.org/docs/man1.0.2/apps/x509v3_config.html">OpenSSL X509 V3 Configuration</a>
+     * @see <a href="https://en.wikipedia.org/wiki/SubjectAltName">Subject Alternative Names on Wikipedia'</a>
+     */
+    public static GeneralName makeGeneralName(String name) {
+        Validate.isTrue(name != null, "Certificate general name cannot be null");
+
+        assert name != null;
```

Comment:
Agreed. I just modeled it after the previous method that was already there. I will remove it as I don't like that anyway.